core/src/main/scala/za/co/absa/spline/core/harvester/componentCreators.scala

```diff
@@ -110,7 +111,7 @@ object ExpressionConverter {
     })
   }
 
-  private[this] def renderValue(recursion: Any => Option[Any])(o: Any): Option[Any] = {
+  private[this] def renderValue(recursion: Any => Option[Any], t : Option[org.apache.spark.sql.types.DataType] = None)(o: Any): Option[Any] = {
```

Comment:
Prefer adding an appropriate prefix or suffix to an optional valuable. E.g. 'maybeType', 'typeOption' etc.